cdap-app-templates/cdap-etl/cdap-data-pipeline/src/main/java/io/cdap/cdap/datapipeline/FieldLineageProcessor.java

```diff
@@ -67,6 +67,7 @@ public FieldLineageProcessor(PipelineSpec pipelineSpec) {
 
     // validate the stage operations
     Map<String, InvalidFieldOperations> stageInvalids = new HashMap<>();
+    Map<String, Map<String, List<String>>> stageRedundants = new HashMap<>();
```

Comment:
should we use another class similar to `InvalidFieldOperations` to follow the same pattern? This seem to be exposing the ds directly.

Reply:
`InvalidFieldOperations` is just a wrapper of two `Map<String, List<String>>` of input and output fields. Here we don't need that as we only need one map. The ds will not get exposed. The map is like <stageName, <field, List<operation name>>>, which is basically is same as `InvalidFieldOperations`